dbt-adapters/src/dbt/include/global_project/macros/materializations/functions/function.sql

```diff
@@ -0,0 +1,16 @@
+{% materialization function, default %}
+    {% set existing_relation = load_cached_relation(this) %}
+    {% set target_relation = this.incorporate(type=this.Function) %}
+
+    {{ run_hooks(pre_hooks) }}
+
+    {% set function_type_macro = get_function_macro('scalar', 'sql') %}
```

Comment:
We should stop hard coding these variables once we start supporting other function types + languages